From 0f73054810284ffbc2de990438df1655bf5232ed Mon Sep 17 00:00:00 2001
From: Pavel Zwerschke <pavel.zwerschke@quantco.com>
Date: Wed, 28 Oct 2026 13:03:25 +0100
Subject: [PATCH] Use bundled queue.h on Windows

Windows does not ship <sys/queue.h>, which is needed by the public
lshpack.h header. Include the upstream compat copy under a unique name
(lshpack_queue.h) so it can be installed without clobbering a generic
sys/queue.h from other packages.
---
 lshpack.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lshpack.h b/lshpack.h
index 80cd763..b844c5b 100644
--- a/lshpack.h
+++ b/lshpack.h
@@ -28,7 +28,11 @@ SOFTWARE.
 
 #include <limits.h>
 #include <stdint.h>
+#ifdef _WIN32
+#include "lshpack_queue.h"
+#else
 #include <sys/queue.h>
+#endif
 #include "lsxpack_header.h"
 
 #ifdef __cplusplus
-- 
2.56.0

